src/app/pages/statistics-page/statistics.page.ts

```diff
@@ -33,56 +46,111 @@ export class StatisticsPage implements OnInit, OnDestroy {
+        ({ periodRange, monthControl, yearControl }) => {
+          this.selectedPeriod = periodRange;
+          let startDate = 0;
+          let endDate = 0;
+          if (periodRange === ViewPeriod.Month) {
+            const currentMonth = this.translateService.instant(
+              'general.months.' + monthControl
+            );
+            if (currentMonth.includes('.')) {
+              startDate = moment(getCurrentYear() + monthControl, DateFormatEnum.YYYYMMMM)
+                .startOf(ViewPeriod.Month)
+                .unix();
+              endDate = moment(getCurrentYear() + monthControl, DateFormatEnum.YYYYMMMM)
+                .endOf(ViewPeriod.Month)
+                .unix();
+            } else {
+              startDate = moment(
+                getCurrentYear() + 
+                this.translateService.instant('general.months.' + monthControl),
+                DateFormatEnum.YYYYMMMM
+              )
+                .startOf(ViewPeriod.Month)
+                .unix();
+              endDate = moment(
+                getCurrentYear() +
+                this.translateService.instant('general.months.' + monthControl),
+                DateFormatEnum.YYYYMMMM
+              )
+                .endOf(ViewPeriod.Month)
+                .unix();
+            }
+          } else if (periodRange === ViewPeriod.Year) {
+            startDate = moment(String(yearControl), DateFormatEnum.YYYY)
+              .startOf(ViewPeriod.Year)
+              .unix();
+            endDate = moment(String(yearControl), DateFormatEnum.YYYY)
+              .endOf(ViewPeriod.Year)
+              .unix();
           } else {
-            startDate = moment(
-              getCurrentYear() + this.translateService.instant('general.months.' + monthControl), DateFormatEnum.YYYYMMMM
-            ).startOf(ViewPeriod.Month).unix();
-            endDate = moment(
-              getCurrentYear() + this.translateService.instant('general.months.' + monthControl), DateFormatEnum.YYYYMMMM
-            ).endOf(ViewPeriod.Month).unix();
+            startDate = moment().startOf(periodRange).unix();
+            endDate = moment().endOf(periodRange).unix();
           }
-        } else if (periodRange === ViewPeriod.Year) {
-          startDate = moment(String(yearControl), DateFormatEnum.YYYY).startOf(ViewPeriod.Year).unix();
-          endDate = moment(String(yearControl), DateFormatEnum.YYYY).endOf(ViewPeriod.Year).unix();
-        } else {
-          startDate = moment().startOf(periodRange).unix();
-          endDate = moment().endOf(periodRange).unix();
-        }
 
-        this.store.dispatch(SpendingActions.updateStatTimePeriod({ payload: { startDate, endDate } }));
-        this.updateList();
-      })
+          this.store.dispatch(SpendingActions.updateStatTimePeriod({
+              payload: { startDate, endDate },
+            })
+          );
+          this.updateList();
+        }
+      )
     );
```

Comment:
a lot of logic in OnInit, make sense split these on several methods and call them in oninit (functional approuch)